component/prometheus/exporter/cloudwatch/config_test.go

```diff
@@ -0,0 +1,45 @@
+package cloudwatch
+
+import (
+	"github.com/grafana/agent/pkg/river"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func Test(t *testing.T) {
+	config := `
+sts_region = "us-east-2"
+debug = true
+discovery {
```

Comment:
@cristiangreco @kgeckhart what do you think of adding also a label here in the `discovery` block which define a discovery job in YACE? That way they have a "name" like static ones, and we either ignore it (it's just for clarity in the river code), or add it as a custom tag in the job's YACE config

It will look like: 
```hcl
discovery "queues" {
	type = "sqs"
	regions = ["us-east-2"]
	search_tags = {
		"scrape" = "true",
	}
	metric {
		name = "NumberOfMessagesReceived"
		statistics = ["Sum", "Average"]
		period = "1m"
	}
}
```

and we can either ignore it, or add it as `custom-tag-name="queues"`. The main purpose of this is to be consistent with static jobs that do require a "name".

Reply:
I think we can leave it off or make it optional. YACE doesn't allow naming the discovery jobs and I think the name will largely be overlapping with the `type`. If we thinks it's valuable though I think it should come from the YACE config.

Reply:
oki doki, leaving it off then